### PR TITLE
ARC-1457 file content lambda validation

### PR DESCRIPTION
**Pull request by atahmasebiupgrade**
Changes to virus scanning lambda code so that it verifies the file type is allowed by looking at its content.

### CODE REVIEW

File: scan.py [hunk 1]
```diff
@@ -142,6 +143,37 @@ def sns_scan_results(s3_object, result):
...
+    print("comparing s3_content_type=[%s] vs magic_content_type=[%s]" % (s3_object.content_type, content_type))
+    return content_type is not None and content_type == s3_object.content_type
+
+
+def is_file_content_type_allowed(file_path):
+    allowed_mime_types = ["application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+                          "application/vnd.ms-excel",
+                          "image/gif",
+                          "image/png",
+                          "image/jpeg",
```
Comment (dfosticupgrade) [inline scan.py]: what about "image/jpg"?

File: scan.py [hunk 2]
```diff
@@ -142,6 +143,38 @@ def sns_scan_results(s3_object, result):
...
 
+def is_content_type_match_file_content(s3_object, file_path):
+    content_type = magic.from_file(file_path, mime=True)
+    print("comparing s3_content_type=[%s] vs magic_content_type=[%s]" % (s3_object.content_type, content_type))
+    return content_type is not None and content_type == s3_object.content_type
+
+
+def is_file_content_type_allowed(file_path):
+    allowed_mime_types = ["application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+                          "application/vnd.ms-excel",
```
Comment (pgangodkerupgrade) [inline scan.py]: Why do we need to allow excel files? @akulshreshthaupgrade and @eviquezcredify  - can we see if we can get rid of this even for internal file uploads?
Also - @atahmasebiupgrade  - would you please check to see that we have a validation at borrower portal orch api level to block borrowers from uploading excel files? @eviquezcredify  mentioned we used to only allow VQ users to upload excel but given changes over time would be good to verify.

Comment (atahmasebiupgrade) [reply scan.py]: As of R38, the validation between borrower-portal-orch and vq-orch are going to be the same. 
https://github.com/Credify/shared-lib/blob/master/credify-validator/src/main/java/com/credify/request/validator/ValidBorrowerDocumentExtension.java

I was trying to keep the validation consistent between vq-orch and borrower-dashboard-orch so we don't have to maintain them separately. I was hopoing to work with Andy and Alok to see if we can remove excel from both sides.

Comment (dfosticupgrade) [reply scan.py]: @atahmasebiupgrade i just tried to upload .xls, .xlt & .xlsx and they go to infected
so should be able to upload or not?

Comment (atahmasebiupgrade) [reply scan.py]: As of now, `.xls` and `.xlsx` should not be marked as `INFECTED` as long as they are genuine excel documents. `.xlt` should be marked as infected though. 

I'm creating another ticket to remove `.xls` and `.xlsx` from the list of allowed extensions.

**Comment by jpbelangerupgrade** [comment]
Why would a generic AV scan tool do business validation for file type. This makes no sense at all.. This should have one and only one purpose:
Scan for Virus

no virus, it goes thru...
virus it returns an error. Validation mime type should be done at the application side. 

We're really starting to put logic in this module and that is really bad. This binds us to code. Instead of being one day able to change this solution easily since it does only one job.

**Comment by atahmasebiupgrade** [comment]
@jpbelangerupgrade This is not the longterm solution. This was supposed to be a quick PR to stop the bleeding. see: https://github.com/Credify/bucket-antivirus-function/pull/11/files#diff-af7685086e279fe5d5c283ac51ab7f41R166

I also added some application level validation that will go live with the next release here:
https://github.com/Credify/shared-lib/pull/846

**Pull request merged**